### PR TITLE
Implement the REST portions of Admin Portal 2.0

### PR DESCRIPTION
**Pull request by hawkowl**
### Which issue this PR addresses:

See https://github.com/Azure/ARO-RP/pull/1676

See also ADO-9777019

### What this PR does / why we need it:

Implements the REST and unit test portions of the Admin Portal REST API.

### Test plan for issue:

Unit tests

### Is there any documentation that needs to be updated for this PR?

Includes some for making more tests

### CODE REVIEW

File: cmd/aro/portal.go [hunk 1]
```diff
@@ -158,7 +158,7 @@ func portal(ctx context.Context, log *logrus.Entry, audit *logrus.Entry) error {
 		return err
 	}
 
-	hostname := "localhost:8444"
+	hostname := "localhost:3000"
```
Comment (Makdaam) [inline cmd/aro/portal.go:161]: nit: hostname vs. address could use a comment
Took me a while to make sure that hostname is the externally visible part of the URI (used for AAD redirects for now).

File: pkg/portal/cluster/clusteroperators.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+type OperatorInformation struct {
+	Name      string                   `json:"name"`
+	Available configv1.ConditionStatus `json:"available"`
+}
+
+type ClusterOperatorsInformation struct {
+	Operators []OperatorInformation `json:"operators"`
+}
+
+func ClusterOperatorsInformationFromClusterDoc(operators *configv1.ClusterOperatorList) *ClusterOperatorsInformation {
```
Comment (Makdaam) [inline pkg/portal/cluster/clusteroperators.go:22]: naming nit: Shouldn't "FromClusterDoc" be "FromOperatorList"?

File: cmd/aro/portal.go (hunk 1 above)
Comment (hawkowl) [reply cmd/aro/portal.go:163]: added a comment that should explain why it's different

File: pkg/portal/cluster/clusteroperators.go (hunk 2 above)
Comment (hawkowl) [reply pkg/portal/cluster/clusteroperators.go]: yep

**Review by darthhexx — DISMISSED** [review]
/lgtm

**Comment by jewzaam** [comment]
Looks good to me, but need to hold on the current RP release, starting today.  Then we can get it merged. Soon!!!

**Review by jewzaam — APPROVED** [review]

**Pull request merged**